implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -74,6 +76,16 @@ public DynatraceExporterV2(DynatraceConfig config, Clock clock, HttpSender httpC
         metricBuilderFactory = factoryBuilder.build();
     }
 
+    private static String makeConcatenatedUriString(String baseUri, String extraPath) throws URISyntaxException, MalformedURLException {
+        // makes sure the base uri is parsable as URL (includes "http://" etc.)
+        URL url = new URL(baseUri);
+        URI uri = url.toURI();
+        // replace all occurrences of two or more forward slashes with a single one
+        String newPath = (uri.getPath() + '/' + extraPath).replaceAll("/{2,}", "/");
+        URI newUri = uri.resolve(newPath).normalize();
+        return newUri.toString();
+    }
+
     static String prepareEndpoint(String uri) {
```

Comment:
I think these attempts to autocomplete the endpoint if only the host is specified are not really obvious. Please make sure that the documentation for the config options points out that this is done.

Reply:
Sounds good to me. Users will have to adapt their config anyway to switch to v2, so I think we can require them to supply the exact endpoint URI here and don't need to autocomplete it for backwards-compatibility. This might be a bit fragile anyway and could break if someone would use a custom proxy for channeling the request to DT, for example (e.g. "https://localhost:12345/dt-endpoints/metrics" or "https://dtrouter.example.com/metrics"). It's probably better to not apply too much magic here.

Reply:
I agree that using the real URI is preferred here. Completing the endpoint URI automatically might result in new corner cases that would need to be handled separately, complicating the logic without much benefit for the user.